politeiawww/cmsuser_test.go

```diff
@@ -0,0 +1,247 @@
+package main
+
+import (
+	"encoding/hex"
+	"testing"
+
+	cms "github.com/decred/politeia/politeiawww/api/cms/v1"
+	www "github.com/decred/politeia/politeiawww/api/www/v1"
+	"github.com/decred/politeia/politeiawww/cmd/shared"
+)
+
+func TestInviteNewUser(t *testing.T) {
+	p, cleanup := newTestCMSwww(t)
+	defer cleanup()
+
+	var tests = []struct {
+		name      string
+		email     string
+		wantError error
+	}{
+		{
+			"success",
+			"test@example.com",
+			nil,
+		},
+		{
+			"error malformed",
+			"testemailmalformed",
+			www.UserError{
+				ErrorCode: www.ErrorStatusMalformedEmail,
+			},
+		},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			inviteUserReq := cms.InviteNewUser{
+				Email:     v.email,
+				Temporary: false,
+			}
+			_, err := p.processInviteNewUser(inviteUserReq)
+
+			got := errToStr(err)
+			want := errToStr(v.wantError)
+			if got != want {
+				t.Errorf("got error %v, want %v", got, want)
+			}
+		})
+	}
+}
+
+func TestRegisterUser(t *testing.T) {
+	p, cleanup := newTestCMSwww(t)
+	defer cleanup()
+
+	// Create user identity and save it to disk
+	id, err := shared.NewIdentity()
+	if err != nil {
+		t.Fatalf("error generating identity")
+	}
+
+	// Create another user identity and save it to disk
+	idFresh, err := shared.NewIdentity()
+	if err != nil {
+		t.Fatalf("error another generating identity")
+	}
+	email := "test1@example.org"
+	username := "test1"
+	pwd := "password1"
+
+	inviteUserReq := cms.InviteNewUser{
+		Email:     email,
+		Temporary: false,
+	}
+	reply, err := p.processInviteNewUser(inviteUserReq)
+	if err != nil {
+		t.Fatalf("error inviting user %v", err)
+	}
+	var tests = []struct {
```

Comment:
Missing test for:
- `ErrorStatusInvalidPublicKey`: for full coverage this should have a case for each of the ways a pubkey can be invalid.
- `ErrorStatusMalformedUsername`: a test case for each way a username can be invalid.
- `ErrorStatusMalformedPassword`
- `ErrorStatusVerificationTokenInvalid`: a test case for both ways a verification token can be invalid
- `ErrorStatusVerificationTokenExpired`

Before I look at the rest of the tests, take these concepts and apply them to all the routes you're adding tests to in this PR.